morello/system_config/cpu.py

```diff
@@ -232,6 +235,16 @@ def delete(self):
         shutil.rmtree(self.whole_dir)
 
 
+class X86Target(_CpuTarget):
+    def _clang_vec_flags(self) -> Sequence[str]:
+        return ["-fopenmp", "-mavx2"]
+
+
+class ArmTarget(_CpuTarget):
+    def _clang_vec_flags(self) -> Sequence[str]:
+        return []
```

Comment:
@ken-matsui Does ARM/M1 not need any compiler flags to use the vector extensions from arm_neon.h?

Additionally, we probably want -fopenmp here unless it's not supported on M1s for some reason.

Reply:
@samkaufman At least M1 didn't need flags as shown in this https://github.com/samkaufman/morello/pull/6#issuecomment-1550545981. I will investigate whether the use of `-fopenmp` results in any compilation errors.

Reply:
Apple Clang didn't recognize the flag, but Homebrew Clang recognized it. (Also, I found Homebrew Clang works with this PR)

```
clang: error: unsupported option '-fopenmp'
```